src/patchelf.cc

```diff
@@ -1082,6 +1082,15 @@ void ElfFile<ElfFileParamNames>::modifySoname(sonameMode op, const std::string &
 
     debug("new SONAME is '%s'\n", newSoname.c_str());
 
+    /* If new SONAME fits in existing space, overwrite. This works around
+       issue #44. */
+    if (newSoname.size() <= sonameSize) {
+       debug("overwriting old SONAME with new...\n");
+       strcpy(soname, newSoname.c_str());
+       changed = true;
+       return;
```

Comment:
This functionality will have to change now @reidpr 
I introduced new changes that always do `this->rewriteSections();` after each `changed = true` This is to support multiple commands on a single invocation.

I agree that this is not an ideal tracking each `return`.
I would like to figure out how to do a destructor for each method and then apply `this->rewriteSections();` only if `changed == true`.

Either way this code will produce errors for now if done with other commands.

Reply:
I opened https://github.com/NixOS/patchelf/pull/363/files# to help make this simpler.

Reply:
Hello, thanks for the heads up. I still hope (after 3 years) this PR will get merged, but I don't have time to maintain it.